.gitbook/contents/developer/boba-basics/basic-ops.md

```diff
@@ -193,41 +159,28 @@ async transfer(address, value_Wei_String, currency) {
 	console.log(' completed Deposit! L2 tx hash:', l2Receipt.transactionHash)
 
 	return l2Receipt
-    
+
   }
 ```
 
 <figure><img src="../../../assets/accessing latest L1 block number.png" alt=""><figcaption></figcaption></figure>
 
-The hex value that corresponds to the `L1BLOCKNUMBER` opcode (`0x4B`) may be changed in the future. **We strongly discourage direct use of this opcode within your contracts.** Instead, if you want to access the latest L1 block number, please use the `OVM_L1BlockNumber` contract as described below.
-
-The block number of the latest L1 block seen by the L2 system can be accessed via the `L1BLOCKNUMBER` opcode. Solidity doesn't make it easy to use non-standard opcodes, so there is a simple contract located at \[`0x4200000000000000000000000000000000000013` that will allow you to trigger this opcode. You can use this contract as follows:
+Information about the L1 is available via the [L1 Block Attributes Predeploy
+Contract](https://github.com/ethereum-optimism/specs/blob/298e745a7aa2e0aa78b6f18eb88bf1c484c00c4a/specs/protocol/deposits.md#l1-attributes-predeployed-contract).  This contract is always updated by the first implicit deposit transaction in every block.
 
 ```javascript
-import { iOVM_L1BlockNumber } from "@eth-optimism/contracts/L2/predeploys/iOVM_L1BlockNumber.sol";
-import { Lib_PredeployAddresses } from "@eth-optimism/contracts/libraries/constants/Lib_PredeployAddresses.sol";
+import { L1Block } from "@eth-optimism/contracts-bedrock/L2/L1Block.sol";
+import { Predeploys } from "@eth-optimism/contracts-bedrock/libraries/Predeploys.sol";
 
 contract MyContract {
    function myFunction() public {
       // ... your code here ...
 
-      uint256 l1BlockNumber = iOVM_L1BlockNumber(
-         Lib_PredeployAddresses.L1_BLOCK_NUMBER // located at 0x4200000000000000000000000000000000000013
+      uint256 l1BlockNumber = L1Block(
+         Predeploys.L1_BLOCK_ATTRIBUTES
       ).getL1BlockNumber();
```

Comment:
```suggestion
      ).number();
```
this seems to be the new method for this
or 
we could also use the same method with the legacy contract